identity/src/commonMain/kotlin/com/android/identity/nfc/NfcIsoTag.kt

```diff
@@ -0,0 +1,218 @@
+
+    /**
+     * Selects a file according to ISO 7816-4 clause 11.2.2.
+     *
+     * @param fileId the identifier for the file to select, e.g. [Nfc.NDEF_CAPABILITY_CONTAINER_FILE_ID].
+     */
+    suspend fun selectFile(fileId: Int) {
+        //
+        val response = transceive(
+            CommandApdu(
+                cla = 0,
+                ins = Nfc.INS_SELECT,
+                p1 = Nfc.INS_SELECT_P1_FILE,
+                p2 = Nfc.INS_SELECT_P2_FILE,
+                payload = ByteString(encodeShort(fileId)),
+                le = 0
+            )
+        )
+        if (response.status != Nfc.RESPONSE_STATUS_SUCCESS) {
+            throw IllegalStateException("Error selecting file, status ${response.statusHexString}")
+        }
+    }
+
+    /**
+     * Reads binary data according to ISO 7816-4 clause 11.3.3.
+     *
+     * @param offset offset of where to read from.
+     * @param length amount of data to read, must be positive.
+     * @return the data which was read.
+     */
+    suspend fun readBinary(offset: Int, length: Int): ByteArray {
+        require(length > 0) { "Length must be positive" }
+        val response = transceive(
+            CommandApdu(
+                cla = 0,
+                ins = 0xb0,
+                p1 = offset / 0x100,
+                p2 = offset.and(0xff),
+                payload = ByteString(),
+                le = length
+            )
+        )
+        if (response.status != Nfc.RESPONSE_STATUS_SUCCESS) {
+            throw IllegalStateException("Error READ BINARY, status ${response.statusHexString}")
+        }
+        return response.payload.toByteArray()
+    }
+
+    /**
+     * Updates binary data according to ISO 7816-4 clause 11.3.5.
+     *
+     * @param offset offset of where to update data.
+     * @param data the data to write, cannot be larger than 255 bytes.
+     */
+    suspend fun updateBinary(offset: Int, data: ByteArray) {
+        require(data.size > 0) { "Data to write must be non-empty" }
+        require(data.size < 256) { "Data cannot be larger than 255 bytes" }
+
+        //
+        val response = transceive(
+            CommandApdu(
+                cla = 0,
+                ins = 0xd6,
+                p1 = offset / 0x100,
+                p2 = offset.and(0xff),
+                payload = ByteString(data),
+                le = 0
+            )
+        )
+        if (response.status != Nfc.RESPONSE_STATUS_SUCCESS) {
+            throw IllegalStateException("Error UPDATE BINARY, status ${response.statusHexString}")
+        }
+    }
+
+    /**
+     * Reads NDEF data according to NFC Forum Type 4 Tag section 7.5.4.
+     *
+     * @param wtInt Minimum waiting time as per NFC Forum Tag NDEF Exchange Protocol section 4.1.6.
+     * @param nWait Maximum number of waiting time extensions as per NFC Forum Tag NDEF Exchange Protocol section 4.1.7.
+     * @return the message that was read.
+     */
+    suspend fun ndefReadMessage(
+        wtInt: Int = 0,
```

Comment:
Do we need to use the WtInt format for this? The API would be easier to use if we had the caller specify a Duration...
And if we switch this, maybe rename the parameters? minWaitingTime? maxWaitExtensions?

Reply:
The original API in `VerificationHelper` was actually like this, it took a `Float`... but the caller will always have `wtInt` and `nWait` at their disposal in the `ServiceParameterRecord` in the initial NDEF message. We also provide `Duration.Companion.fromWtInt()` to convert it to a `Duration`...

As for renaming, I think it's helpful to use the same names as in the NDEF spec...